include/coap2/option.h

```diff
@@ -105,50 +107,49 @@ typedef struct coap_opt_filter_t {
 #define COAP_OPT_ALL NULL
 
 /**
- * Clears filter @p f.
+ * Clears filter @p filter.
  *
- * @param f The filter to clear.
+ * @param filter The filter to clear.
  */
-COAP_STATIC_INLINE void
-coap_option_filter_clear(coap_opt_filter_t *f) {
-  memset(f, 0, sizeof(coap_opt_filter_t));
-}
+void
+coap_option_filter_clear(coap_opt_filter_t *filter);
 
 /**
- * Sets the corresponding entry for @p type in @p filter. This
+ * Sets the corresponding entry for @p option in @p filter. This
```

Comment:
Fair point.  Reverted `option` back to `type` in `option.h`

Reply:
I have thought further about this - the real issue for me is the overloaded use of `type `causing code reading confusion (is this message type, code type etc.). RFC7252 refers to CoAP Option Numbers and the CoAP Option Number Registry, so have gone through and renamed (the new) `coap_pdu_option_t` to `coap_option_num_t` and used the word `number `instead of `type` to better aid code readability.  Updated code pushed.

Reply:
Sounds good, thanks.